x-pack/plugins/ml/server/models/job_validation/validate_cardinality.js

```diff
@@ -65,10 +65,17 @@ const validateFactory = (callWithRequest, job) => {
               });
             }
           } else {
-            messages.push({
-              id: 'field_not_aggregatable',
-              fieldName: uniqueFieldName
-            });
+            // when the job is using categorization and the field name is set to 'mlcategory',
+            // then don't report the field as not being able to be aggregated.
+            if (!(
+              typeof job.analysis_config.categorization_field_name !== 'undefined' &&
+              uniqueFieldName === 'mlcategory'
```

Comment:
the `mlcategory` string is repeated a lot throughout our code base. It would probably be better as a constant.
Out of scope for this PR, just mentioning it.